tests/test_catchmod.py

```diff
@@ -242,6 +246,34 @@ def test_run_catchmod(leap):
         shape[0] -= num_leap
     rainfall = np.ones(shape)
     pet = np.zeros(shape)
-    flow = run_catchmod(catchment, rainfall, pet, dates)
+    flow = run_catchmod(catchment, rainfall, pet, dates, total_flows)
+
     assert(flow.shape[0] == len(dates))
-    assert(flow.shape[1] == rainfall.shape[1])
+    if total_flows:
+        assert(len(flow.shape) == 2)
+        assert (flow.shape[1] == rainfall.shape[1])
+    else:
+        assert(len(flow.shape) == 3)
+        assert (flow.shape[2] == len(catchment.subcatchments))
+        assert (flow.shape[1] == rainfall.shape[1])
+
+
+def test_run_catchmod_subcatchment_flows():
+    """ test the subcatchment flows returned sum to the returned total flows """
+
+    catchment = catchment_from_json(os.path.join(os.path.dirname(__file__), "data", "thames.json"), n=2)
+    dates = pandas.date_range("1920-01-01", periods=200, freq="D")
+
+    rainfall = np.random.randint(0, 20, [200, 2])
+    pet = np.random.randint(0, 5, [200, 2])
+
+    flow_total = run_catchmod(catchment, rainfall, pet, dates, output_total=True)
+    flow_subcatchment = run_catchmod(catchment, rainfall, pet, dates, output_total=False)
+
+    np.testing.assert_equal(flow_total, flow_subcatchment.sum(axis=2))
```

Comment:
We should use `np.testing.assert_allclose` here because these are floats. This might fail exact equality due to floating point precision.